### PR TITLE
*: remove v0manifest

### LINKED ISSUE #906: Remove v0manifest
The `v0manifest` was our initial repository manifest implementation, as we have switched to TUF implementation `v1manifest` since `v1.0.0`, the `v0manifest` is deprecated and only kept for backward compatibility.

The `v1.0.0` was released at May 2020, and it would be reasonable to remove the legacy `v0manifest` codes from `v1.4` and later releases.

Some helper utils and tools used for v0 -> v1 migration should also be removed.

### PR DESCRIPTION
**Pull request by AstroProfundis**
### What problem does this PR solve? 
Close #906 

### What is changed and how it works?
Remove `v0manifest` and legacy codes using it.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change

Side effects

 - Breaking backward compatibility

Release notes:

```release-note
Remove `v0manifest` as it is not longer been used.
```

### CODE REVIEW

File: pkg/environment/env.go [hunk 1]
```diff
@@ -368,28 +256,6 @@ func ParseCompVersion(spec string) (string, pkgver.Version) {
 
 // IsSupportedComponent return true if support if platform support the component.
 func (env *Environment) IsSupportedComponent(component string) bool {
```
Comment (lucklove) [inline pkg/environment/env.go:258]: Should we delete this function?

File: pkg/environment/env.go [hunk 2]
```diff
@@ -99,38 +96,19 @@ func InitEnv(options repository.Options) (*Environment, error) {
 		return nil, err
 	}
 
-	var repo *repository.Repository
 	var v1repo *repository.V1Repository
```
Comment (nrc) [inline pkg/environment/env.go:99]: should we rename `v1repo` to `repo`? And likewise `v1Repo` and `V1Repository`

Comment (nrc) [reply pkg/environment/env.go:99]: and V1Manifest, etc.

**Review by nrc — APPROVED** [review]
lgtm

File: pkg/environment/env.go (hunk 2 above)
Comment (AstroProfundis) [reply pkg/environment/env.go:99]: I think it's OK, as we don't guarantee there won't be `v2repo` in the future. The only misleading thing might be that we still use `v0.1.0` as `spec_version` in `v1Manifest`.

**Pull request merged**